libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -29,78 +32,125 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
+ * ideally will be exported by '@openfeature/web-sdk'
+ * */
+enum ProviderStatus {
+  NOT_READY = 'NOT_READY',
+  READY = 'READY',
+  ERROR = 'ERROR',
+}
+
 export class LaunchDarklyClientProvider implements Provider {
   readonly metadata: ProviderMetadata = {
-    name: 'launchdarkly-client-provider',
+    name: 'launchdarkly-client',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
+
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  get status() {
+    return this._status;
+  }
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  constructor(private readonly envKey: string, { providerLogger, ...ldOptions }: LaunchDarklyProviderOptions) {
+    this.ldOptions = ldOptions;
+    if (providerLogger) {
+      this.logger = providerLogger;
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
   }
 
-  async initialize(context: EvaluationContext): Promise<void> {
-    const newContext = this.translateContext(context);
-    const oldContext = this.client.getContext();
-    const ignoreAnonymous = newContext.anonymous && oldContext.anonymous && !newContext.key
-    if(!ignoreAnonymous && !isEqual(newContext, oldContext)) {
-       await this.client.identify(newContext);
+  private get client(): LDClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
+  }
+
+  async initialize(): Promise<void> {
+    /* for simplicity always initialize with anonymous by default*/
+
+    this._client = initialize(this.envKey, { anonymous: true }, this.ldOptions);
+
+    /*
+     * set event listeners to propagate ld events to open feature provider,
+     * use LD property streaming, since when enabled you are opting in for straming updates
+     * */
+    if (this.ldOptions?.streaming) {
+      this.setListeners();
+    }
+
+    try {
+      await this._client.waitForInitialization();
+      this.status = ProviderStatus.READY;
+    } catch {
+      this.status = ProviderStatus.ERROR;
     }
-    return this.client.waitUntilReady();
   }
 
   onClose(): Promise<void> {
     return this.client.close();
   }
 
+  /** set listeners to LD client and event the correspodent event in the Provider
+   * necessary for LD streaming changes
+   * */
+  private setListeners() {
+    this.client.on('change', (changeset: LDFlagChangeset) => {
+      this.events.emit(ProviderEvents.ConfigurationChanged, {
+        flagKeysChanged: Object.keys(changeset),
```

Comment:
```suggestion
        flagsChanged: Object.keys(changeset),
```

This resolves a compilation issue for me (and the CI) which I think is related to the other issues mentioned in this PR. I _think_ an old version of the web-sdk is being cached _somewhere_ in your workspace :grimacing: